applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClientCredentials.java

```diff
@@ -0,0 +1,140 @@
+    private final CredentialInformation credentials;
+    private final AuthorizationHeaderProvider headerProvider;
+
+    /**
+     * Creates an instance that is able to authorize requests to Azure Application Configuration service.
+     *
+     * @param connectionString connection string in the format "endpoint={endpoint_value};id={id_value};secret={secret_value}"
+     */
+    public ConfigurationClientCredentials(String connectionString) throws InvalidKeyException, NoSuchAlgorithmException {
+        credentials = new CredentialInformation(connectionString);
+        headerProvider = new AuthorizationHeaderProvider(credentials);
+    }
+
+    URL baseUri() {
+        return this.credentials.baseUri;
+    }
+
+    @Override
+    public Mono<String> authorizationHeaderValueAsync(HttpRequest request) {
+        return Mono.just(headerProvider.getAuthenticationHeaderValue(request));
+    }
+
+    private static class AuthorizationHeaderProvider {
+        private final String[] signedHeaders = new String[]{HOST_HEADER, DATE_HEADER, CONTENT_HASH_HEADER};
+        private final String signedHeadersValue = String.join(";", signedHeaders);
+        private final CredentialInformation credentials;
+        private final Mac sha256HMAC;
+
+        AuthorizationHeaderProvider(CredentialInformation credentials) throws NoSuchAlgorithmException, InvalidKeyException {
+            this.credentials = credentials;
+
+            sha256HMAC = Mac.getInstance("HmacSHA256");
+            sha256HMAC.init(new SecretKeySpec(credentials.secret(), "HmacSHA256"));
+        }
+
+        private String getAuthenticationHeaderValue(final HttpRequest request) {
+            final String stringToSign = getStringToSign(request);
+
+            final String signature = Base64.getEncoder().encodeToString(sha256HMAC.doFinal(stringToSign.getBytes(StandardCharsets.UTF_8)));
+            return String.format("HMAC-SHA256 Credential=%s, SignedHeaders=%s, Signature=%s",
+                    credentials.id(),
+                    signedHeadersValue,
+                    signature);
+        }
+
+        private String getStringToSign(final HttpRequest request) {
+            String pathAndQuery = request.url().getPath();
+            if (request.url().getQuery() != null) {
+                pathAndQuery += '?' + request.url().getQuery();
+            }
+
+            final HttpHeaders httpHeaders = request.headers();
+            final String signed = Arrays.stream(signedHeaders)
+                    .map(httpHeaders::value)
+                    .collect(Collectors.joining(";"));
+
+            // String-To-Sign=HTTP_METHOD + '\n' + path_and_query + '\n' + signed_headers_values
+            // Signed headers: "host;x-ms-date;x-ms-content-sha256"
+            return String.format("%s\n%s\n%s", request.httpMethod().toString().toUpperCase(Locale.US), pathAndQuery, signed);
+        }
+    }
+
+    private class CredentialInformation {
```

Comment:
@conniey just a thought - when I looked closely on this type, this is actually an object representing `ConnectionString` right? so was wondering if so should it be named `ConnectionString` with a static factory method named `parse(string)` that takes connection string in string format and return `ConnectionString`? The usual pattern we see for connection string types.

Reply:
There are various formats for connectionstrings. We have an open issue to try and maybe make a ConnectionString some sort of ConnectionDetail. Or something. https://github.com/Azure/azure-sdk-for-java/issues/3080.

I can change this to be a factory method .parse for now.

Reply:
For now just build what makes sense for Configuration Service, and don't invest too heavily in it - we will be changing this before too long, once the identity and authentication work lands as a common API.